Trying to run MIRI on MacOS

### Steps to Reproduce

* Make a new cargo project
* Add `wasmtime = "*"`
* Do `cargo miri run`

### Expected Results
Miri compiles and runs successfully

### Actual Results
Failed to compile because of something inside the crate itself. I looked into the module traphandlers and sure enough there is no function named that
```
error[E0425]: cannot find function `using_mach_ports` in module `traphandlers`
  --> /Users/macbook/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasmtime-runtime-17.0.1/src/traphandlers.rs:52:23
   |
52 |         traphandlers::using_mach_ports(),
   |                       ^^^^^^^^^^^^^^^^ not found in `traphandlers`

For more information about this error, try `rustc --explain E0425`.
```

### Versions and Environment

Wasmtime version or commit: 17.0.1

Operating system: MacOS

Architecture: ARM

--- a/crates/runtime/src/sys/miri/traphandlers.rs
+++ b/crates/runtime/src/sys/miri/traphandlers.rs
@@ -41,6 +41,7 @@ pub fn platform_init(_macos_use_mach_ports: bool) {}
 
 pub fn lazy_per_thread_init() {}
 
+#[cfg(target_os = "macos")]
 pub fn using_mach_ports() -> bool {
     false
 }

--- a/crates/runtime/src/sys/miri/traphandlers.rs
+++ b/crates/runtime/src/sys/miri/traphandlers.rs
@@ -40,3 +40,7 @@ pub type SignalHandler<'a> = dyn Fn() + Send + Sync + 'a;
 pub fn platform_init(_macos_use_mach_ports: bool) {}
 
 pub fn lazy_per_thread_init() {}
+
+pub fn using_mach_ports() -> bool {
+    false
+}

--- a/crates/wasmtime/src/config.rs
+++ b/crates/wasmtime/src/config.rs
@@ -244,7 +244,7 @@ impl Config {
             force_memory_init_memfd: false,
             wmemcheck: false,
             coredump_on_trap: false,
-            macos_use_mach_ports: true,
+            macos_use_mach_ports: !cfg!(miri),
         };
         #[cfg(any(feature = "cranelift", feature = "winch"))]
         {

